html/browsers/windows/browsing-context-names/browsing-context-default-name.html

```diff
@@ -5,22 +5,25 @@
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <div id="log"></div>
-<iframe src="message.html" style="display:none"></iframe>
+<iframe src="/common/blank.html" style="display:none"></iframe>
+<object id="obj" type="text/html" data="about:blank"></object>
+<embed id="embedded" type="image/svg+xml" src="/images/green.svg" width="0" height="0" />
 <script>
+test(t => {
+  assert_equals(window.frames[0].name, "");
+  assert_equals(document.getElementById("embedded").name, "");
+  assert_equals(window["obj"].name, "");
+}, "A embedded browsing context has empty-string default name");
 
-test(function () {
-  assert_equals(window.frames[0].name, "", "The browsing context should not have a default name.");
-}, "A embedded browsing context has no default name");
-
-test(function () {
+test(t => {
   var win = window.open("about:blank", "_blank");
-  assert_equals(win.name, "", "The browsing context should not have a name.");
+  assert_equals(win.name, "");
   win.close();
-}, "A browsing context which is opened by window.open() method with '_blank' parameter has no default name");
+}, "A browsing context which is opened by window.open() method with '_blank' parameter has empty-string default name");
 
 //This test must be run when the current browsing context's name is not set
-test(function () {
-  assert_equals(window.name, "", "The browsing context should not have a name.");
-}, "A browsing context has no default name");
+test(t => {
```

Comment:
Side note: This (pre-existing to me) test will fail when run with the web-based runner because the test's browsing context has some artifacts from the test runner (e.g. has a `name`, `opener`). I will track down how to handle this kind of situation separately...